server/src/main/java/org/apache/druid/server/coordinator/helper/DruidCoordinatorSegmentInfoLoader.java

```diff
@@ -43,6 +43,12 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
   {
     log.info("Starting coordination. Getting available segments.");
 
+    final Iterable<DataSegment> dataSegments = coordinator.iterateAvailableDataSegments();
+    if (dataSegments == null) {
+      log.info("Metadata store not polled yet, canceling this run.");
```

Comment:
I think maybe using 'delay' instead of 'cancel', maybe something like "delaying segment coordination" or something of that sort would read better in logs.